python/src/opendp/mod.py

```diff
@@ -787,27 +787,28 @@ def binary_search_param(
     return binary_search(lambda param: make_chain(param).check(d_in, d_out), bounds, T)
 
 @overload
-def binary_search( # type: ignore[overload-overlap]
-        predicate: Callable[[T], bool],
-        bounds: Optional[Tuple[T, T]] = None,
-        T: Optional[Type[T]] = None,
-        return_sign: Literal[False] = False) -> T:
+def binary_search(
+        predicate: Callable[[float], bool],
+        bounds: Tuple[float, float] | None = ...,
+        T: Type[float] | None = ...,
+        return_sign: Literal[False] = False) -> float:
     ...
 
 
 @overload
 def binary_search(
-        predicate: Callable[[T], bool],
-        bounds: Optional[Tuple[T, T]] = None,
-        T: Optional[Type[T]] = None,
-        return_sign: Literal[True] = True) -> Tuple[T, int]:
+        predicate: Callable[[float], bool],
+        bounds: Tuple[float, float] | None = ...,
+        T: Type[float] | None = ...,
+        *,
```

Comment:
```suggestion
```

I'm expecting the parameter lists of the typing overloads to align with the final definition?

Reply:
It's in a weird spot behind optional params, and gets hit with this issue: https://stackoverflow.com/questions/66435480/overload-following-optional-argument

I've added an extra overload.